02-express-tutorial/controllers/people.js

```diff
@@ -0,0 +1,61 @@
+const { people } = require("../data");
+
+function addPerson(req, res) {
+  const { id, name } = req.body;
+  const newPerson = { id, name };
+  people.push(newPerson);
+  res.status(201).json(newPerson);
+}
+
+function getPeople(req, res) {
+  res.status(200).json(people);
+}
+
+function getPersonById(req, res) {
+  const id = parseInt(req.params.id);
+  const person = people.find((p) => p.id === id);
+
+  if (person) {
+    res.status(200).json(person);
+  } else {
+    res.status(404).json({ message: "Person not found" });
+  }
+}
+
+function updatePerson(req, res) {
+  const id = parseInt(req.params.id);
+  const personIndex = people.findIndex((p) => p.id === id);
+
+  if (personIndex !== -1) {
+    // Update the person entry
+    people[personIndex] = { ...people[personIndex], ...req.body };
+    res.status(200).json(people[personIndex]);
+  } else {
+    res.status(404).json({ message: "Person not found" });
+  }
```

Comment:
Nitpick (doesn't require changes):
This can be simplified to using `Array.find`:
```js
const id = parseInt(req.params.id, 10);
const person = people.find(person => person.id === personId);

if (!person) {
  return res.status(404).json({ message: "Person not found" });
} 

person.name = req.body.updatedName; // or whatever property you're using to update the person property

// or if you're update multiple properties...
// person = { ...person, ...req.body };

res.status(200).json(person);
```